core/trino-main/src/main/java/io/trino/execution/SqlTask.java

```diff
@@ -586,6 +584,41 @@ public SqlTaskIoStats getIoStats()
             TaskContext taskContext = taskExecution.getTaskContext();
             return new SqlTaskIoStats(taskContext.getProcessedInputDataSize(), taskContext.getInputPositions(), taskContext.getOutputDataSize(), taskContext.getOutputPositions());
         }
+
+        public VersionedDynamicFilterDomains acknowledgeAndGetNewDynamicFilterDomains(long callersSummaryVersion)
+        {
+            // if we are finished, return the final VersionedDynamicFilterDomains
+            if (finalDynamicFilterDomains != null) {
+                return finalDynamicFilterDomains;
+            }
+            // if we haven't started yet, return an empty VersionedDynamicFilterDomains
+            if (taskExecution == null) {
+                return INITIAL_DYNAMIC_FILTER_DOMAINS;
+            }
+            // get VersionedDynamicFilterDomains from the current task execution
+            TaskContext taskContext = taskExecution.getTaskContext();
+            return taskContext.acknowledgeAndGetNewDynamicFilterDomains(callersSummaryVersion);
+        }
+
+        public long getDynamicFiltersVersion()
+        {
+            // if we are finished, return the final VersionedDynamicFilterDomains
+            if (finalDynamicFilterDomains != null) {
+                return finalDynamicFilterDomains.getVersion();
+            }
+            requireNonNull(taskExecution, "taskExecution is null");
+            return taskExecution.getTaskContext().getDynamicFiltersVersion();
+        }
+
+        public VersionedDynamicFilterDomains getFinalDynamicFilterDomains()
```

Comment:
the name of this method is confusing as it suggest it would return value fo `finaleDynamicFilterDomains` field. 
I suggest you drop it and inline in the only place where it is used.

Reply:
I've renamed this now, PTAL
I've trying to keep it similar to `getIoStats`